examples/with-fiber/main.go

```diff
@@ -0,0 +1,159 @@
+					   need to configure a Service ID on the Apple developer dashboard and use that as client ID.
+					   In the example below 4398792-io.supertokens.example.service is the client ID for Web. Android etc and thus we mark it as default. For iOS
+					   the frontend for the demo app sends the clientId in the request which is then used by the SDK.
+					*/
+					thirdparty.Apple(tpmodels.AppleConfig{
+						// For Android and website apps
+						IsDefault: true,
+						ClientID:  "4398792-io.supertokens.example.service",
+						ClientSecret: tpmodels.AppleClientSecret{
+							KeyId:      "7M48Y4RYDL",
+							PrivateKey: "-----BEGIN PRIVATE KEY-----\nMIGTAgEAMBMGByqGSM49AgEGCCqGSM49AwEHBHkwdwIBAQQgu8gXs+XYkqXD6Ala9Sf/iJXzhbwcoG5dMh1OonpdJUmgCgYIKoZIzj0DAQehRANCAASfrvlFbFCYqn3I2zeknYXLwtH30JuOKestDbSfZYxZNMqhF/OzdZFTV0zc5u5s3eN+oCWbnvl0hM+9IW0UlkdA\n-----END PRIVATE KEY-----",
+							TeamId:     "YWQCXGJRJL",
+						},
+					}),
+					thirdparty.Apple(tpmodels.AppleConfig{
+						// For iOS Apps
+						ClientID: "4398792-io.supertokens.example",
+						ClientSecret: tpmodels.AppleClientSecret{
+							KeyId:      "7M48Y4RYDL",
+							PrivateKey: "-----BEGIN PRIVATE KEY-----\nMIGTAgEAMBMGByqGSM49AgEGCCqGSM49AwEHBHkwdwIBAQQgu8gXs+XYkqXD6Ala9Sf/iJXzhbwcoG5dMh1OonpdJUmgCgYIKoZIzj0DAQehRANCAASfrvlFbFCYqn3I2zeknYXLwtH30JuOKestDbSfZYxZNMqhF/OzdZFTV0zc5u5s3eN+oCWbnvl0hM+9IW0UlkdA\n-----END PRIVATE KEY-----",
+							TeamId:     "YWQCXGJRJL",
+						},
+					}),
+				},
+			}),
+			session.Init(nil),
+			emailpassword.Init(nil),
+		},
+	})
+	if err != nil {
+		log.Fatal("Something went wrong while starting up supertokens: ", err.Error())
+	}
+	app := fiber.New()
+
+	//adding the supertokens middleware
+	app.Use(adaptor.HTTPMiddleware(supertokens.Middleware))
+
+	allowedHeaders := append([]string{"Content-Type"}, supertokens.GetAllCORSHeaders()...)
+	allowedHeadersInCommaSeparetedStringFormat := stringArrayToStringConvertor(allowedHeaders)
+
+	app.Use(cors.New(cors.Config{
+		AllowOrigins:     "http://localhost:3000",
+		AllowMethods:     "GET, POST, PUT, HEAD, OPTIONS",
+		AllowHeaders:     allowedHeadersInCommaSeparetedStringFormat,
+		AllowCredentials: true,
+	}))
+
+	app.Get("/sessInfo", verifySession(nil), sessioninfo)
+	log.Fatal(app.Listen(":3001"))
+}
+
+//wrapper of the original implementation of verify session to match the required function signature
+func verifySession(options *sessmodels.VerifySessionOptions) fiber.Handler {
+	return func(c *fiber.Ctx) error {
+		adaptor.HTTPHandler(session.VerifySession(options, func(rw http.ResponseWriter, r *http.Request) {
+			//setting up the verified session context from http request to the fiber context
+			c.SetUserContext(r.Context())
+		}))(c)
+		sessionContainer := session.GetSessionFromRequestContext(c.UserContext())
+		if sessionContainer != nil {
+			return c.Next()
+		} else {
```

Comment:
It should call `c.Next()` even if sessionContainer is nil and `*options.SessionRequired == true`